cmd/agent/agent.go

```diff
@@ -86,12 +93,24 @@ func start(cmd *cobra.Command, args []string) {
 		log.Fatal("Failed to create the agent: ", err)
 	}
 
+	consultemplateconfig := &agent.ConsulTemplateConfig{
+		Source:      templatePath,
+		Destination: fmt.Sprintf("%s/%s/%s", configDir, serviceName, METADATAFILE),
+	}
+	runner, err := agent.StartConsulTemplate(a, consultemplateconfig)
+	if err != nil {
+		log.Fatal("could not start consul template: ", err)
+	}
+
 	go func() {
 		quit := <-signals
 		log.Printf("Caught %s signal!", quit)
 
 		log.Println("Stopping the agent...")
 		a.Stop()
+
+		// Stop consul-template
+		agent.StopConsulTemplate(runner)
```

Comment:
Same goes here: the CLI command handler is the only consumer of `Agent` public API, and from this standpoint nothing should have changed: we still start and stop just the `a` instance.